src/main/java/org/jboss/ejb/client/Affinity.java

```diff
@@ -117,11 +129,24 @@ public String toString() {
     static class LocalAffinity extends Affinity {
 
         private static final long serialVersionUID = -2052559528672779420L;
+        private static final URI uri;
+
+        static {
+            try {
+                uri = new URI("local", "-", null);
```

Comment:
I'm not super happy about this.  But Java doesn't support conforming "empty" URIs with only a scheme part.  Still thinking of a way to sling this weakness into an advantage, somehow.